checkov/secrets/git_history_store.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import logging
+import copy
+from collections import defaultdict
+from typing import TYPE_CHECKING, Dict, List, Tuple, Optional
+from typing_extensions import TypedDict
+from checkov.secrets.consts import GIT_HISTORY_NOT_BEEN_REMOVED
+
+if TYPE_CHECKING:
+    from detect_secrets.core.potential_secret import PotentialSecret
+
+
+class EnrichedPotentialSecret(TypedDict):
+    added_commit_hash: str
+    removed_commit_hash: str
+    potential_secret: PotentialSecret
+    code_line: Optional[str]
+
+
+class GitHistorySecretStore:
+    def __init__(self) -> None:
+        self.secrets_by_file_value_type: Dict[str, List[EnrichedPotentialSecret]] = {}
+
+    def set_secret_map(self, file_results: List[PotentialSecret],
+                       file_name: str, commit_hash: str, commit: Dict[str, str | Dict[str, str]]) -> None:
+        # First find if secret was moved in the file
+        equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
+        for secret in file_results:
+            secret_key = get_secret_key(file_name, secret.secret_hash, secret.type)
+            equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
+
+        for secret in file_results:
+            if secret.filename in ['None', '']:
```

Comment:
```suggestion
            if not secret.filename:
```

Reply:
that won't work sadly.. I'm getting the string 'None' from detect secrets so 'if not' is no good here

Reply:
ah got it, didn't see it is a string 🙈 then make it at least a set and make it global const or class attribute